addon/components/-private/base-pop-menu.js

```diff
@@ -251,9 +298,89 @@ export default class BasePopMenuComponent extends Component {
+   */
+  _popperKeyHandler = (event) => {
+    let keyCode = event.key;
+    let focusableElements = this._getFocusableElementsInPopper();
+    let activeItem = document.activeElement;
+
+    if (keyCode === 'Escape') {
+    // Escape provides a way to close the popper while the user is focused anywhere
+    // inside the popper. We assume upon closing in this way, the user wants to
+    // be focused where they were before (which would be the trigger).
+      this._closePopoverAndFocusTrigger();
+    } else if (keyCode === 'Enter' && event.target.hasAttribute('data-close')) {
+      // When focus is on a data-close element, pressing Enter should close the popper.
+      // We don't want to make any assumptions about where the focus should go from
+      // here because the app instead should determine where it goes depending on the action.
+      // TODO Ember 1.13: this needs to close both subdropdown and parent dropdown
+      this._closePopoverHandler();
+    } else if (keyCode === 'Tab' && activeItem === focusableElements[focusableElements.length - 1]) {
+      // Since the user has hit the Tab key on the last item in the popper, their
+      // intent is to continue on in the DOM. So we manually focus back to where
+      // they were before opening the popper (which would be the trigger).
+      // The browser will execute the focus on the trigger first, then the tab event
+      // will execute, thus naturally landing on the next tabbable item in the DOM after the trigger.
+      this._closePopoverAndFocusTrigger();
+    } else if (event.shiftKey && keyCode === 'Tab' && activeItem === focusableElements[0]) {
+      // Since the user has hit the shift+Tab key on the first item in the popper, their
+      // intent is to go backwards in the DOM. So we manually focus back to where
+      // they were before opening the popper (which would be the trigger).
+      // The browser will execute the focus on the trigger first, then the shift+tab event
+      // will execute, thus naturally landing on the previous tabbable item in the DOM before the trigger.
+      this._closePopoverAndFocusTrigger();
+    }
+  };
+
+  /**
+   * Maintains a determinable list of elements that are focusable.
+   * Makes a query for elements that match that list
+   * TODO: Make this a global helper?
+   * @return {Array} list of focusable elements
+   */
+  _getFocusableElementsInPopper = () => {
+    let focusableSelectors = 'a[href]:not([disabled]), button:not([disabled]), input:not([disabled]), select:not([disabled]), textarea:not([disabled]), [tabindex]:not([disabled]):not([tabindex="-1"])';
+    let focusableElements = this._popperElement.querySelectorAll(focusableSelectors);
+    return focusableElements;
+  };
+
+  /**
+   * Gets the list of focusable elements in the popper, focuses on the first one.
+   * Makes sure there is actually something to focus on.
+   * TODO: Make this a global helper?
+   */
+  _focusOnFirstFocusableElement = () => {
```

Comment:
Looks like this one should also be a standard class method, not an event listener